subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/SyncTaskIntegrationTest.groovy

```diff
@@ -394,84 +394,83 @@ class SyncTaskIntegrationTest extends AbstractIntegrationSpec {
         buildFile << '''
             task syncIt(type: Sync) {
-                from project.hasProperty("capitalizeFile") ? "FILE.TXT" : "file.txt"
-                from project.hasProperty("capitalizeDir") ? "DIR" : "dir"
+                from project.hasProperty("capitalize") ? "FILE.TXT" : "file.txt"
                 into buildDir
             }
         '''
-
-        def lowercaseFile = file('file.txt')
-        lowercaseFile.text = 'file'
-
-        file('dir').create {
-            file('file2.txt') << 'file2'
-            file('file3.txt') << 'file3'
-            file('file4.txt') << 'file4'
-            nestedDir {
-                file('nestedDirFile.txt') << 'nestedDirFile'
-            }
-        }
         and:
         run 'syncIt'
-        file('build').assertHasDescendants(
-            'file.txt',
-            'file2.txt',
-            'file3.txt',
-            'file4.txt',
-            'nestedDir/nestedDirFile.txt'
-        )
+        file('build/file.txt').assertExists()
 
         and:
-        def uppercaseFile = new File(lowercaseFile.parentFile, 'FILE.TXT') // make sure case is kept
         lowercaseFile.renameTo(uppercaseFile)
-        uppercaseFile.text = 'FILE'
         assert uppercaseFile.canonicalFile.name == 'FILE.TXT'
-        and:
-        file('dir/file3.txt').delete()
-
-        file('dir/file4.txt').delete()
-        file('dir/File4.txt') << 'File4'
 
         when:
-        succeeds('syncIt', '-PcapitalizeFile')
+        succeeds('syncIt', '-Pcapitalize')
         then:
         executedAndNotSkipped ':syncIt'
         file('build/FILE.TXT').with {
             assert it.parentFile.list() != [].toArray()
             assert it.assertExists()
-            assert it.text == 'FILE'
             assert it.canonicalFile.name == 'FILE.TXT'
         }
-        and:
-        file('build/file2.txt').assertExists()
-        file('build/file3.txt').assertDoesNotExist()
-        file('build/File4.txt').with {
-            it.assertExists()
-            it.canonicalFile.name == 'File4.txt'
-        }
+    }
 
-        when:
+    @Issue("https://github.com/gradle/gradle/issues/9586")
+    def "change in case of input folder will still sync properly"() {
+        given:
+        file('dir').create {
+            file('file2.txt').createFile()
+            file('file3.txt').createFile()
+            file('file4.txt').createFile()
+            nestedDir {
+                file('nestedDirFile.txt').createFile()
+            }
+        }
+        buildFile << '''
+            task syncIt(type: Sync) {
+                from project.hasProperty("capitalize") ? "DIR" : "dir"
+                into buildDir
+            }
+        '''
+        and:
+        run 'syncIt'
+        file('build').assertHasDescendants(
+            'file2.txt',
+            'file3.txt',
+            'file4.txt',
+            'nestedDir/nestedDirFile.txt'
+        )
+        and:
         file('dir').deleteDir()
         file('DIR').create {
```

Comment:
We can't use `renameTo()` here, because there are contents, right?

Reply:
I'll reduce the number of files and we can use `renameTo` then (was easier to delete and recreate previously)

Reply:
Changed:
```
lowercaseDir.deleteDir()
uppercaseDir.create {
    file('file.txt').createFile()
    NESTEDDIR {
        file('nestedDirFile1.txt').createFile()
        file('NESTEDDIRFILE2.TXT').createFile()
    }
}
```
to
```
lowercaseDir.renameTo(uppercaseDir)

def uppercaseNestedDir = new File(uppercaseDir, 'NESTEDDIR')
new File(uppercaseDir, 'nestedDir').renameTo(uppercaseNestedDir)
new File(uppercaseNestedDir, 'nestedDirFile.txt').renameTo(new File(uppercaseNestedDir, 'NESTEDDIRFILE.TXT'))
```